packages/core/src/index.ts

```diff
@@ -213,11 +214,19 @@ export const makeZodI18nMap: MakeZodI18nMap = (option) => (issue, ctx) => {
       );
       break;
     case ZodIssueCode.custom:
-      message = t("errors.custom", {
-        ns,
-        defaultValue: message,
-        ...path,
-      });
+      if (issue.params?.i18nKey !== undefined) {
+        message = t(issue.params?.i18nKey, {
+          ns: customErrorNs,
+          defaultValue: message,
+          ...path,
+        });
+      } else {
+        message = t("errors.custom", {
+          ns,
+          defaultValue: message,
+          ...path,
+        });
+      }
```

Comment:
```suggestion
      message = t(issue.params?.i18n || "errors.custom", {
        ns,
        defaultValue: message,
        ...path,
      });
```

Reply:
how this can work?
does i18next search in the all the namespaces until he find a namespace with the key?

Reply:
Yes, that's right.  
If `nsA` and `nsB` have the same key and the user wants to make the namespace explicit, the key can be `nsB:test_custom_key`.
This library has been created following the i18next specification as much as possible.

In fact, this test succeeds.
```ts
describe("custom error message", () => {
  test("custom error with refine", async () => {
    await i18next.init({
      lng: "en",
      resources: {
        en: {
          zod_custom: {
            test_custom_key: "custom error message",
          },
        },
      },
    });

    z.setErrorMap(makeZodI18nMap({ ns: ["zod", "zod_custom"] })); // <= multi ns

    expect(
      getErrorMessage(
        z
          .string()
          .refine(() => false, { params: { i18n: "test_custom_key" } })
          .safeParse("")
      )
    ).toEqual("custom error message");
  });
});
```